src/util/signature.cr

```diff
@@ -55,7 +55,8 @@ class Dir
   #   - When a file added, moved, removed, renamed (including which in nested
   #       directories)
   def self.contents_signature(dirname, cache = {} of String => String) : String
-    return cache[dirname] if cache[dirname]?
+    return cache[dirname] if !cache.nil? && cache[dirname]?
```

Comment:
`cache`'s type is `{} of String => String` so it's not nilable.

Reply:
but the compiler complains this because of [this line](https://github.com/hkalexling/Mango/blob/44a6f822cd2d69156beb359dfa26b522f47e875d/src/library/title.cr#L22) and [this line].(https://github.com/hkalexling/Mango/blob/44a6f822cd2d69156beb359dfa26b522f47e875d/src/library/title.cr#L35). I'm using the Crystal v1.1.1, could this be the reason?

Reply:
I fixed `Title.new` to have a cache and fixed this